cubi_tk/sodar/ingest.py

```diff
@@ -0,0 +1,366 @@
+        source_paths = self.build_file_list()
+        source_paths = self.process_md5_sums(source_paths)
+
+        # Initiate iRODS session
+        irods_session = self._init_irods()
+
+        # Query target collection
+        logger.info("Querying landing zone collections…")
+        collections = []
+        try:
+            coll = irods_session.collections.get(self.lz_irods_path)
+            for c in coll.subcollections:
+                collections.append(c.name)
+        except:
+            logger.error("Failed to query landing zone collections.")
+            sys.exit(1)
+        finally:
+            irods_session.cleanup()
+
+        if self.args.collection is None:
+            user_input = ""
+            input_valid = False
+            input_message = "####################\nPlease choose target collection:\n"
+            for index, item in enumerate(collections):
+                input_message += f"{index+1}) {item}\n"
+            input_message += "Select by number: "
+
+            while not input_valid:
+                user_input = input(input_message)
+                if user_input.isdigit():
+                    user_input = int(user_input)
+                    if 0 < user_input < len(collections):
+                        input_valid = True
+
+            self.target_coll = collections[user_input - 1]
+
+        elif self.args.collection in collections:
+            self.target_coll = self.args.collection
+        else:
+            logger.error("Selected target collection does not exist in landing zone.")
+            sys.exit(1)
+
+        # Create sub-collections for folders
+        if self.args.recursive:
+            dirs = sorted(
+                {p["ipath"].parent for p in source_paths if not p["ipath"].parent == Path(".")}
+            )
+
+            logger.info("Planning to create the following sub-collections:")
+            for d in dirs:
+                print(f"{self.target_coll}/{str(d)}")
+            if not self.args.y:
+                if not input("Is this OK? [y/N] ").lower().startswith("y"):
+                    logger.error("Aborting at your request.")
+                    sys.exit(0)
+
+            for d in dirs:
+                coll_name = f"{self.lz_irods_path}/{self.target_coll}/{str(d)}"
+                try:
+                    irods_session.collections.create(coll_name)
+                except Exception as e:
+                    logger.error("Error creating sub-collection.")
+                    logger.error(e)
+                    sys.exit(1)
+                finally:
+                    irods_session.cleanup()
+            logger.info("Sub-collections created.")
+
+        # Build transfer jobs
+        jobs = self.build_jobs(source_paths)
+
+        # Final go from user & transfer
+        total_bytes = sum([job.bytes for job in jobs])
+        logger.info("Planning to transfer the following files:")
+        for job in jobs:
+            print(job.path_src)
```

Comment:
Should probably user logger not print here

Reply:
Again: I like the way it looks.